dash/include/dash/algorithm/Copy.h

```diff
@@ -493,17 +493,42 @@ GlobOutputIt copy_impl(
                  "l_in_last:",   in_last,
                  "g_out_first:", out_first.pos());
 
+
+
   auto num_elements = std::distance(in_first, in_last);
-  dart_storage_t ds = dash::dart_storage<ValueType>(num_elements);
-  DASH_ASSERT_RETURNS(
-    dart_put_blocking(
-      out_first.dart_gptr(),
-      in_first,
-      ds.nelem,
-      ds.dtype),
-    DART_OK);
 
-  auto out_last = out_first + num_elements;
+  if (num_elements < 1) return out_first;
+
+  auto nremaining = static_cast<dash::default_size_t>(num_elements);
```

Comment:
You are right, it is `ssize_t`. The case I had in mind is the following: the user might decides to compile with the flag `DASH_ENABLE_DEFAULT_INDEX_TYPE_INT` (for whatever reason) and then copy more than 4GB into the array, which causes `num_elements` to be truncated into `nremaining`. He might even pass `size_t` as template parameter `IndexType` to the global data structure so it's a valid operation. I think using the pattern's index-type is the safest bet. But that is certainly a highly pathologic case :)